src/ops.rs

```diff
@@ -1376,3 +1382,136 @@ fn op_resources(
     },
   ))
 }
+
+fn subprocess_stdio_map(v: msg::SubprocessStdio) -> std::process::Stdio {
+  match v {
+    msg::SubprocessStdio::Inherit => std::process::Stdio::inherit(),
+    msg::SubprocessStdio::Piped => std::process::Stdio::piped(),
+    msg::SubprocessStdio::Null => std::process::Stdio::null(),
+  }
+}
+
+fn op_run(
+  state: &Arc<IsolateState>,
+  base: &msg::Base,
+  data: &'static mut [u8],
+) -> Box<Op> {
+  assert!(base.sync());
```

Comment:
@piscisaureus Note that it is fundamentally sync. I don't think it blocks tho. Can you elaborate on your problem with it being sync?

Reply:
Ok, I see what you mean. Yes, in order to support "dispatchv" we would need to allocate RIDs on the JS side. I'm ok with that. However there are many places currently where we're allocating RIDs and I think we should change them all to be allocated in JS at once. Therefore I'd argue in favor of keeping this as it is for now.